### ISSUE TITLE
ES Modules are in Safari TP

### ISSUE THREAD

**Issue by surma** [Fyrd/caniuse#5897]
I tried to find the raw data in the JSON for [`mdn-api_worker_worker_ecmascript_modules`](https://caniuse.com/mdn-api_worker_worker_ecmascript_modules), but wasn’t successful.

The table currently reads “unknown”, but support was added in Safari TP 122 ([announcement](https://webkit.org/blog/11577/release-notes-for-safari-technology-preview-122/)). 

Where/How can I fix that?

**Comment by Schweinepriester** [comment]
TL;DR: https://github.com/mdn/browser-compat-data/blob/main/api/Worker.json#L138 :)

---

Hi @surma! 🤩

That feature is coming from <https://github.com/mdn/browser-compat-data> (BCD).

The fastest way to get to the data from [caniuse](https://caniuse.com/mdn-api_worker_worker_ecmascript_modules) is to check the tab Feedback of a given feature, so in this case:

> the [support data for this feature](https://github.com/mdn/browser-compat-data/blob/main/api/Worker.json).

=> Starting at this line: https://github.com/mdn/browser-compat-data/blob/main/api/Worker.json#L138

Sadly BCD doesn't track Safari TP (yet?): https://github.com/mdn/browser-compat-data/issues/4840#issuecomment-603289685 :/

However!
Per <https://webkit.org/blog/11648/new-webkit-features-in-safari-14-1/#:~:text=availability> 

> Changes in this release of Safari were included in the following Safari Technology Preview releases: [110](https://webkit.org/blog/10929/release-notes-for-safari-technology-preview-110/), [111](https://webkit.org/blog/10967/release-notes-for-safari-technology-preview-111/), [112](https://webkit.org/blog/11183/release-notes-for-safari-technology-preview-112/), [113](https://webkit.org/blog/11294/release-notes-for-safari-technology-preview-113/), [114](https://webkit.org/blog/11300/release-notes-for-safari-technology-preview-114/), [115](https://webkit.org/blog/11333/release-notes-for-safari-technology-preview-115/), [116](https://webkit.org/blog/11348/release-notes-for-safari-technology-preview-116/), [117](https://webkit.org/blog/11364/release-notes-for-safari-technology-preview-117/), [118](https://webkit.org/blog/11439/release-notes-for-safari-technology-preview-118/), [119](https://webkit.org/blog/11525/release-notes-for-safari-technology-preview-119/), [120](https://webkit.org/blog/11548/release-notes-for-safari-technology-preview-120/), [121](https://webkit.org/blog/11555/release-notes-for-safari-technology-preview-121/), [122](https://webkit.org/blog/11577/release-notes-for-safari-technology-preview-122/).

support should also be in Safari 14.1 (including iOS 14.5) 🎉
So you could change the data to show support starting with those versions.

Do you have a test at hand? If you want me to I could check Safari 14.1 and iOS 14.5 :)

**Comment by meliixo** [comment]
Safari is iPhone so that's my mother's phone how to get rid of it.?

**Comment by surma** [comment]
@Schweinepriester Thanks for the guidance!

I don’t have a test at hand, but I did check WPT, where [Safari TP is passing the tests but Safari 14.1 is not](https://wpt.fyi/results/workers/modules/dedicated-worker-import.any.html?label=master&product=chrome%5Bexperimental%5D&product=edge%5Bexperimental%5D&product=safari%5Bstable%5D&product=safari%5Bexperimental%5D&aligned)

**Comment by Schweinepriester** [comment]
Bummer! Looking at the changesets and issues of 

> -   Added support for modules in Workers and Worklets ([r273203](https://trac.webkit.org/changeset/273203/webkit/))
> -   Added support for modules in Service Workers ([r273224](https://trac.webkit.org/changeset/273224/webkit/))

I couldn't find a flag or discussion to keep it in TP only or not let it ride with 14.1.

Tried digging around a bit in the mirrored source at https://github.com/WebKit/WebKit, but didn't come to a firm assertion, since I couldn't figure out which branch actually is 14.1.

But I guess the changes weren't ported to 14.1 and will only come with the next version, be that 14.2 or 15.

If that's the case I'm afraid we have to wait for that next version, unless you manage to convince BCD to track Safari TP…

**Comment by surma** [comment]
Just to double-check: There’s no way to set the flag for Safari TP (which currently reads “unknown”) unless BCD starts tracking it?

**Comment by Fyrd** [comment]
Funny you should bring this up, since a PR just got made at BCD to add support for Safari TP (and other preview browsers) :
https://github.com/mdn/browser-compat-data/pull/10334
So should be able to update it there soon!

As for overriding, not a bad idea. Assuming that would be specific for browsers tracked on caniuse and not BCD, ideally there shouldn't be a need to override support data. :)

**Comment by Schweinepriester** [comment]
Damn, I should have been able to find that…

Great news though! :)
Once preview browsers are available the desire to override indeed largely vanishes I think.
